### PR TITLE
1811 - Add KEDA Update Mutation (v0.16)

### PR DESCRIPTION
**Pull request by samblackk**
## Description

Add's a mutation that allows user to enable or disable the KEDA worker in LTS 0.16

## 🎟 Issue(s)

Backports changes made to resolve https://github.com/astronomer/issues/issues/1811

## 🧪 Functional Testing

Can test in playground: 
```
mutation updateDeploymentKedaConfig(
  $deploymentUuid: Uuid!
  $state: Boolean!
) {
  updateDeploymentKedaConfig(
    deploymentUuid: $deploymentUuid
    state: $state
  ) {
    id
    config
  }
}
```

**Variables:**
`deploymentUuid` = deployment.id
`state` = true/false boolean

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `yarn lint && yarn test` before taking out of draft
- [x] Added/updated applicable tests
- [ ] Tested against [Astro-UI](https://github.com/astronomer/astro-ui/) and [Astro-CLI](https://github.com/astronomer/astro-cli/) clients.
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

**Comment by adam2k** [comment]
> @adam2k Could I get a review on this ASAP? Needs to be merged by EOD.

The code looks good.  Give me a minute to functional test it at least in the playground.

**Review by schnie — APPROVED** [review]

**Comment by adam2k** [comment]
I'm getting a `"Cannot set property 'keda' of undefined"` Error with a k8s deployment:

**Deployment Response**
```
{
  "data": {
    "workspaceDeployments": [
      {
        "id": "ckgtmp9ag007x0774shtcjfpl",
        "label": "Test this",
        "description": "",
        "type": "airflow",
        "releaseName": "ecliptical-quasar-6331",
        "version": "0.0.0",
        "workspace": {
          "id": "ckgtmp2r7006s0774ikvq04bi"
        },
        "urls": [
          {
            "type": "airflow",
            "url": "https://deployments.local.astronomer.io/ecliptical-quasar-6331/airflow"
          }
        ],
        "createdAt": "2020-10-28T16:44:08.389Z",
        "updatedAt": "2020-10-28T16:52:52.823Z",
        "config": {
          "executor": "KubernetesExecutor"
        },
        "environmentVariables": [],
        "properties": {
          "extra_au": 10,
          "component_version": "0.0.0",
          "alert_emails": []
        }
      }
    ]
  }
}
```

**Error in the Keda Mutation**
```
{
  "errors": [
    {
      "message": "Cannot set property 'keda' of undefined",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "updateDeploymentKedaConfig"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "TypeError: Cannot set property 'keda' of undefined",
            "    at AuthDirective.updateDeploymentKedaConfig (/Users/adamvandover/Documents/GitHub/snyk-houston-api/src/resolvers/mutation/update-deployment-keda-config/index.js:23:3)",
            "    at runMicrotasks (<anonymous>)",
            "    at processTicksAndRejections (internal/process/task_queues.js:97:5)"
          ]
        }
      }
    }
  ],
  "data": {
    "updateDeploymentKedaConfig": null
  }
}
```

File: src/resolvers/mutation/update-deployment-keda-config/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  // Throw error if deployment doesn't exist
+  if (!deployment) throw new InvalidDeploymentError();
+
+  // Update KEDA config object
+  const config = deployment.config;
+  config.workers = config.workers || {};
+  config.workers.keda = {
+    enabled: state
+  };
```
Comment (adam2k) [inline src/resolvers/mutation/update-deployment-keda-config/index.js:26]: This fix corrects the bug that I was running into in the playground.  Thanks!

Nit pick:

```
  import { set } from "lodash";
  ...
  const config = deployment.config;
  set(config, "workers.keda.enabled", state);
```

**Review by adam2k — APPROVED** [review]
LGTM.  The Bug is resolved!  🎉 🌮

**Comment by samblackk** [comment]
@adam2k Can I merge this even though `scan-trivy` failed?

**Comment by adam2k** [comment]
> @adam2k Can I merge this even though `scan-trivy` failed?

Yeah, that's a known issue.  This is good to merge.

**Pull request merged**